testsuite/objects/__init__.py

```diff
@@ -73,12 +73,14 @@ def __post_init__(self):
 class Credentials:
     """Dataclass for Credentials structure"""
 
-    in_location: str
+    in_location: Literal["authorizationHeader", "customHeader", "queryString", "cookie"]
     keySelector: str
 
     def asdict(self):
-        """Custom asdict, because I cannot use 'in' as a name"""
-        return {"in": self.in_location, "keySelector": self.keySelector}
+        """Custom asdict because of needing to put location as parent dict key for inner dict"""
+        if self.in_location == "authorizationHeader":
```

Comment:
In this case I do not think Abstract class is required but yes

Reply:
I was thinking about this for some time but cant find any solution that fits better as just custom `asdict` as its in commit https://github.com/Kuadrant/testsuite/pull/248/commits/88fed67d671a04712841c950522b540914ab8408. Please take a look at my new commit https://github.com/Kuadrant/testsuite/pull/248/commits/4673136b28eac5af9b0e3c294397e841fa299eb4 as it is my best idea and if you think its okay I will do tests refactor. Otherwise please advise how would you like this to look.

Reply:
I do not think the assert is needed, I would simply remove the value from in_header Literal to indicate what values are accepted in the main class

Reply:
I reverted, please make your refactor PR as we discussed. Thanks!